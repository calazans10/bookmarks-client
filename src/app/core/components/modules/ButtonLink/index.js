import React from 'react';
import PropTypes from 'prop-types';
import { Button } from './styles';

const ButtonLink = ({ children, onClick }) => (
  <Button type="button" onClick={onClick}>
    {children}
  </Button>
);

ButtonLink.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default ButtonLink;
