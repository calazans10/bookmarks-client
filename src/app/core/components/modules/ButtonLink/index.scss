@import 'styles/utilities';

.button-link {
  padding: 0;
  transition: color 0.1s linear;
  border: none;
  background-color: transparent;
  background-image: none;
  color: #2980b9;
  text-decoration: none;
  white-space: nowrap;
  cursor: pointer;
  touch-action: manipulation;
  user-select: none;

  &:focus,
  &:hover {
    color: #20638f;
  }
}
