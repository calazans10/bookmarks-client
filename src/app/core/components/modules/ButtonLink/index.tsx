import React from 'react';
import { Button } from './style';

type ButtonLinkProps = {
  children: React.ReactNode;
  onClick: () => void;
};

const ButtonLink = ({ children, onClick }: ButtonLinkProps) => (
  <Button type="button" onClick={onClick}>
    {children}
  </Button>
);

export default ButtonLink;
