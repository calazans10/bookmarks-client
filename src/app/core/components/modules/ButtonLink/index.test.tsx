import React from 'react';
import { render } from '@testing-library/react';
import ButtonLink from './';

describe('ButtonLink', () => {
  it('renders without crashing', () => {
    const { container } = render(<ButtonLink onClick={jest.fn()}>Edit</ButtonLink>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
