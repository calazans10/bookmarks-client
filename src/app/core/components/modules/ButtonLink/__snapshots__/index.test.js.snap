// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders without crashing 1`] = `
<button
  class="sc-AykKC hSJRhy"
  type="button"
>
  Edit
</button>
`;
